test/suite/test_error_info.py

```diff
@@ -26,15 +26,75 @@
 
+import errno
 import wiredtiger, wttest, time
 from wtdataset import SimpleDataSet
+from compact_util import compact_util
+
+import unittest
 
 # test_error_info.py
 #   Test that the placeholder get_last_error() session API returns placeholder error values.
-class test_error_info(wttest.WiredTigerTestCase):
+class test_error_info(compact_util):
+    table_name1 = 'table:test_error'
 
-    def test_error_info(self):
+    def check_error(self, error, sub_level_error, error_msg):
         err, sub_level_err, err_msg = self.session.get_last_error()
-        self.assertEqual(err, 0)
-        self.assertEqual(sub_level_err, wiredtiger.WT_NONE)
-        self.assertEqual(err_msg, "")
+        self.assertEqual(err, error)
+        self.assertEqual(sub_level_err, sub_level_error)
+        self.assertEqual(err_msg, error_msg)
+
+    def test_compaction_already_running(self):
+        # Create a basic large table.
+        SimpleDataSet(self, self.table_name1, 1000, key_format='S', value_format='S').populate()
+
+        # Enable the background compaction server.
+        self.turn_on_bg_compact()
+
+        # Attempt to reconfigure.
+        self.assertRaisesException(wiredtiger.WiredTigerError, lambda: self.session.compact(None, f'background=true,free_space_target=10MB'))
+
+        # Expect error code, sub-error code and error message to reflect compaction already running.
+        self.check_error(errno.EINVAL, wiredtiger.WT_COMPACTION_ALREADY_RUNNING, "Cannot reconfigure background compaction while it's already running.")
+
+    def test_uncommitted_data(self):
+        # Create a simple table.
+        self.session.create(self.table_name1, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.table_name1)
+
+        # Start a transaction and insert a key and value.
+        self.session.begin_transaction()
+        cursor.set_key('key')
+        cursor.set_value('value')
+        self.assertEqual(cursor.update(), 0)
+        cursor.close()
+
+        # Attempt to drop the table without committing the transaction.
+        self.assertRaisesException(wiredtiger.WiredTigerError, lambda: self.session.drop(self.table_name1, None))
+
+        # Expect error code, sub-error code and error message to reflect uncommitted data.
+        self.check_error(errno.EBUSY, wiredtiger.WT_UNCOMMITTED_DATA, "the table has uncommitted data and can not be dropped yet")
+
+    def test_dirty_data(self):
+        # Create a simple table.
+        self.session.create(self.table_name1, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.table_name1)
+
+        # Start a transaction, insert a key and value, and commit the transaction.
+        self.session.begin_transaction()
+        cursor.set_key('key')
+        cursor.set_value('value')
+        self.assertEqual(cursor.update(), 0)
+        self.assertEqual(self.session.commit_transaction(), 0)
+        cursor.close()
+
+        time.sleep(1)
```

Comment:
Do you need the sleep?

Reply:
It is unexpected since you do call `commit_transaction` just above.

Reply:
After some testing, it looks like [this check](https://github.com/wiredtiger/wiredtiger/blob/98a9357bf5d6e77444e39b4876b00643214fc13e/src/include/txn_inline.h#L944-L945) returns `false` when you don't have the sleep. When you have the sleep, everything is visible and the [next check](https://github.com/wiredtiger/wiredtiger/blob/98a9357bf5d6e77444e39b4876b00643214fc13e/src/include/txn_inline.h#L948-L949) returns true.

Reply:
[Here](https://github.com/wiredtiger/wiredtiger/blob/98a9357bf5d6e77444e39b4876b00643214fc13e/src/include/txn_inline.h#L902-L904), without the sleep, the `oldest_id` stays at 2 while your transaction id 5 is greater, which makes it seems not everything is visible.
When you add the sleep, `oldest_id` is bumped to something greater than your transaction, 6 in my case. And the test returns that everything is visible.

Reply:
As discussed on Zoom, another test does it, see [here](https://github.com/wiredtiger/wiredtiger/pull/10905/files#diff-e50a9a7285d87f7ba21cab0f5bbb0ddc9376c428ba98f2a3588fa20c161894b0R96-R98). We asked the team to understand the motivation behind this. It should not block the PR as I now believe it's an expected behaviour.